File: src/Homework1/Homework1.Tests/ProgramTests.cs
using Microsoft.VisualStudio.TestPlatform.TestHost;
using Xunit;

namespace Homework1.Tests
{
    public class ProgramTests
    {
        [Theory]
        [InlineData(new string[] {"2", "+", "3"}, 0)]
        [InlineData(new string[] {"2", "*", "3"}, 0)]
        [InlineData(new string[] {"2", "-", "1"}, 0)]
        [InlineData(new string[] {"2", "/", "2"}, 0)]
        public void Main_ValidArgumentsWorksAsExcepted(string[] args, int excepted)
        {
            var actual = Program.Main(args);
            Assert.Equal(excepted, actual);
        }
        
        [Theory]
        [InlineData(new string[] {".", "+", "."}, 1)]
        [InlineData(new string[] {"2", ".", "3"}, 2)]
        [InlineData(new string[] {".", ".", "."}, 1)]
        [InlineData(new string[] {"", "", ""}, 1)]
        public void Main_InvalidArgumentsWorksAsExcepted(string[] args, int excepted)
        {
            var actual = Program.Main(args);
            Assert.Equal(excepted, actual);
        }
    }
}

File: tests/HW5.Tests/ProgramTests.fs
module HW5.Tests.ProgramTests

open HW5
open Xunit
open Program

[<Theory>]
[<InlineData("1", "+", "3")>]
let ``Main valid`` arg1 arg2 arg3 =
    let actual = main [|arg1; arg2; arg3|]
    Assert.Equal(0, actual)
    
[<Theory>]
[<InlineData("1", ".", "3")>]
let ``Main invalid`` arg1 arg2 arg3 =
    let actual = main [|arg1; arg2; arg3|]
    Assert.Equal(1, actual)
